graph/graph.go

```diff
@@ -112,11 +113,18 @@ func NewGraph(root string, driver graphdriver.Driver) (*Graph, error) {
 	}
 
 	graph := &Graph{
-		root:     abspath,
-		idIndex:  truncindex.NewTruncIndex([]string{}),
-		driver:   driver,
-		retained: &retainedLayers{layerHolders: make(map[string]map[string]struct{})},
+		root:             abspath,
+		idIndex:          truncindex.NewTruncIndex([]string{}),
+		driver:           driver,
+		retained:         &retainedLayers{layerHolders: make(map[string]map[string]struct{})},
+		tarSplitDisabled: false,
 	}
+
+	// Windows does not currently support tarsplit functionality.
+	if runtime.GOOS == "windows" {
+		graph.tarSplitDisabled = true
```

Comment:
Not surprised. In the past weeks, I've had had the issue surface a couple
times that the current "diff"ing is rendering useless the benefits of a
couple of the graphdriver.
On Aug 7, 2015 16:37, "Stefan J. Wernli" <notifications@github.com> wrote:

> In graph/graph.go
> <https://github.com/stevvooe/docker/pull/1#discussion_r36558851>:
>
> >  	}
> > +
> > +	// Windows does not currently support tarsplit functionality.
> > +	if runtime.GOOS == "windows" {
> > +		graph.tarSplitDisabled = true
>
> Yeah, we'd certainly appreciate the help. I had an email ready to contact
> you about it, but then we decided it would be better to unblock progress
> for the short term by skipping it since the resolution is likely to be
> complicated. Our transport format (which gets produced during the call to
> Diff) is different than the on-disk layer format. I think that is part of
> the complication.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stevvooe/docker/pull/1/files#r36558851>.
>